syft/pkg/cataloger/java/parse_java_manifest.go

```diff
@@ -115,16 +115,39 @@ func selectName(manifest *pkg.JavaManifest, filenameObj archiveFilename) string
 }
 
 func selectVersion(manifest *pkg.JavaManifest, filenameObj archiveFilename) string {
-	var version string
-	switch {
-	case filenameObj.version != "":
-		version = filenameObj.version
-	case manifest.Main["Implementation-Version"] != "":
-		version = manifest.Main["Implementation-Version"]
-	case manifest.Main["Specification-Version"] != "":
-		version = manifest.Main["Specification-Version"]
-	case manifest.Main["Plugin-Version"] != "":
-		version = manifest.Main["Plugin-Version"]
+	if v := filenameObj.version; v != "" {
+		return v
+	}
+
+	if manifest == nil {
+		return ""
```

Comment:
is this a problem that it's no longer `nil` (I think)?

Reply:
Good question — this was actually returning `""` before, too, it was just implicit. `var version string` sets the new variable (`version`) to the "zero-value" of the type (`string`), which is `""`.

Reply:
Aha! 👍